ddns-scripts: fix detecting local ip from ip command

chris5560 opened the pull request:
Maintainer: me

Description:
fix detecting local ip from ip command #3834 and
https://forum.lede-project.org/t/bugs-in-ddns-scripts/1000

Signed-off-by: Christian Schoenebeck <<christian.schoenebeck@gmail.com>>

mionica commented:
Hi Christian,
not sure that will work - the 'peer A.B.C.D' is left in place, so for an input like
```
10: pppoe-wan: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1492 qdisc fq_codel state UNKNOWN qlen 3\    link/ppp
10: pppoe-wan    inet 86.125.248.7 peer 10.0.0.1/32 scope global pppoe-wan\       valid_lft forever preferred_lft forever
10: pppoe-wan    inet6 2a02:2f09:306f:ffff::567d:f807/128 scope global \       valid_lft forever preferred_lft forever
```
you'd end up with
```
inet 86.125.248.7 peer 10.0.0.1 -1
inet6 2a02:2f09:306f:ffff::567d:f807
```
and the script would get confused; hence the 
```
; s,peer [0-9\./]* ,,g
```
bit I was suggesting

chris5560 commented:
@mionica 
sorry I missed your suggestion
I rewrote completely and split into sed and awk
I hope the relevant ip on ppp connections is listed between inet and peer.
I personally have no experience on ppp connections.
please recheck
many thanks

chris5560 commented:
@hnyman
seems to be fixed. Could you please merge.
Many thanks
Christian
